apps/web/src/lib/csv.ts

```diff
@@ -8,34 +8,39 @@ const parser = new Parser({
   withBOM: true,
 });
 
-export const toCsv = (
-  happening: Exclude<Awaited<ReturnType<typeof getHappeningCsvData>>, undefined>,
-): string => {
+export const toCsv = (happening: Exclude<Awaited<ReturnType<typeof getHappeningCsvData>>, undefined>, selectedHeaders: Array<string>): string => {
   const registrations = happening.registrations.map((r) => {
     const answers = r.answers.map((a) => ({
       questionId: a.questionId,
       question: happening.questions.find((q) => q.id === a.questionId)?.title ?? "Unknown question",
       answer: a.answer?.answer,
     }));
-
     const obj: Record<string, string> = {};
     obj.Epost = r.user.alternativeEmail ?? r.user.email;
     obj.Navn = r.user.name ?? "Ingen navn";
     obj.Status = r.status;
-    // eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing
-    obj.År = r.user.year?.toString() || "Ingen år";
+    obj.År = r.user.year?.toString() ?? "Ingen år";
     obj.Studieretning = r.user.degreeId ?? "Ingen studieretning";
     obj.Grunn = r.unregisterReason ?? "";
 
-    for (const question of happening.questions) {
-      const answer = answers.find((a) => a.questionId === question.id)?.answer;
-
-      const formattedAnswer = Array.isArray(answer) ? answer.join(", ") : answer ?? "";
-
-      obj[question.title] = formattedAnswer;
-    }
+    const filteredObj: Record<string, string> = {};
+    selectedHeaders.forEach((header) => {
+      if (header === "Alternativ Epost") {
+        header = "Epost";
+      }
+      if (Object.prototype.hasOwnProperty.call(obj, header)) {
```

Comment:
Vet ikke om det funker, men hadde ikke (kode under) vært det samme?

```suggestion
      if (header in obj) {
```

Reply:
Jo de funket, nice der gitt eg ned et rabbithole ass